polaris-migrator/src/migrations/replace-sass-spacing/replace-sass-spacing.ts

```diff
@@ -0,0 +1,31 @@
+import {FileInfo} from 'jscodeshift';
+import postcss, {Plugin} from 'postcss';
+
+import {createRegexFromMap} from '../../utilities/regex';
+
+const spacingMap = {
+  'spacing(none)': 'var(--p-space-0)',
+  'spacing(extra-tight)': 'var(--p-space-1)',
+  'spacing(tight)': 'var(--p-space-2)',
+  'spacing(base-tight)': 'var(--p-space-3)',
+  'spacing()': 'var(--p-space-4)',
+  'spacing(base)': 'var(--p-space-4)',
+  'spacing(loose)': 'var(--p-space-5)',
+  'spacing(extra-loose)': 'var(--p-space-8)',
+};
+
+const plugin = (): Plugin => ({
+  postcssPlugin: 'ReplaceSassSpacing',
+  Declaration(decl) {
+    decl.value = decl.value.replace(
+      createRegexFromMap(spacingMap),
```

Comment:
Dragging a comment over from the old PR: https://github.com/Shopify/polaris/pull/6701#discussion_r937015021

Use [`postcss-value-parser`](https://www.npmjs.com/package/postcss-value-parser) here to have more AST focused approach to parsing the value instead of string replacements. We should be able to say "only act when you find a function name, whose name is "spacing".

This will avoid false positives like `my-spacing()`, and catch variations in whitespace/quoting like `spacing( none )`, `spacing('none')` and `spacing( "none" )` (admittedly as we use prettier our whitespace should be pretty conventional so this shouldn't be a huuuge issue for our codebases).


stylelint has a demo of using `postcss-value-parser` https://github.com/stylelint/stylelint/blob/main/lib/rules/function-allowed-list/index.js#L36